lib/rubocop/cop/style/safe_navigation.rb

```diff
@@ -0,0 +1,104 @@
+# encoding: utf-8
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop transforms usages of a method call safeguarded by a call to
+      # `respond_to?` to safe navigation `&.`.
+      #
+      # @example
+      #   # bad
+      #   foo.bar if foo.respond_to?(:bar)
```

Comment:
Shouldn't this be `foo.bar if foo`? We care only about redundant nil checks here after all. 
`foo && foo.bar` is another classic pattern. As is `foo.bar unless foo.nil?`.